src/Core/Grid/Definition/Factory/CatalogPriceRuleGridDefinitionFactory.php

```diff
@@ -0,0 +1,345 @@
+{
+    const GRID_ID = 'catalog_price_rule';
+
+    public function __construct(
+        HookDispatcherInterface $hookDispatcher
+    ) {
+        parent::__construct($hookDispatcher);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function getId()
+    {
+        return self::GRID_ID;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function getName()
+    {
+        return $this->trans('Catalog price rules', [], 'Admin.Catalog.Feature');
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function getColumns()
+    {
+        return (new ColumnCollection())
+            ->add((new BulkActionColumn('bulk'))
+                ->setOptions([
+                    'bulk_field' => 'id_specific_price_rule',
+                ])
+            )
+            ->add((new DataColumn('id_specific_price_rule'))
+                ->setName($this->trans('ID', [], 'Admin.Global'))
+                ->setOptions([
+                    'field' => 'id_specific_price_rule',
+                ])
+            )
+            ->add((new DataColumn('name'))
+                ->setName($this->trans('Name', [], 'Admin.Global'))
+                ->setOptions([
+                    'field' => 'name',
+                ])
+            )
+            ->add((new DataColumn('shop'))
+                ->setName($this->trans('Shop', [], 'Admin.Global'))
+                ->setOptions([
+                    'field' => 'shop',
+                ])
+            )
+            ->add((new DataColumn('currency'))
+                ->setName($this->trans('Currency', [], 'Admin.Global'))
+                ->setOptions([
+                    'field' => 'currency',
+                ])
+            )
+            ->add((new DataColumn('country'))
+                ->setName($this->trans('Country', [], 'Admin.Global'))
+                ->setOptions([
+                    'field' => 'country',
+                ])
+            )
+            ->add((new DataColumn('group'))
+                ->setName($this->trans('Group', [], 'Admin.Global'))
+                ->setOptions([
+                    'field' => 'group_name',
+                ])
+            )
+            ->add((new DataColumn('from_quantity'))
+                ->setName($this->trans('From quantity', [], 'Admin.Catalog.Feature'))
+                ->setOptions([
+                    'field' => 'from_quantity',
+                ])
+            )
+            ->add((new DataColumn('reduction_type'))
+                ->setName($this->trans('Reduction type', [], 'Admin.Catalog.Feature'))
+                ->setOptions([
+                    'field' => 'reduction_type',
+                ])
+            )
+            ->add((new DataColumn('reduction'))
+                ->setName($this->trans('Reduction', [], 'Admin.Catalog.Feature'))
+                ->setOptions([
+                    'field' => 'reduction',
+                ])
+            )
+            ->add((new DateTimeColumn('from'))
```

Comment:
@matks, for this specific page, exists such timestamps '0000-00-00 00:00:00'.
DateTimeColumn renders such time stamp in to strange numbers  (image bellow). Any offers for that?
![Screenshot from 2019-05-15 15-37-59](https://user-images.githubusercontent.com/31609858/57776187-7d3b9f00-7727-11e9-8b80-c9052e49ed50.png).

Reply:
Usually, when `0000-00-00 00:00:00` value is used in a database, it's an alias for `null` but the database does not allow nullable values.

So I suggest to display it like a blank value. Nothing.

@colinegin @MatShir Do you agree ? Display a blank space instead of this irrelevant `-0001-11-30 00:00` ?

@zuk3975 Create a php constant `const DATETIME_NULL_VALUE = '0000-00-00 00:00:00';` to handle it so you can do `if ($date === self::DATETIME_NULL_VALUE)` in the code

Reply:
https://github.com/PrestaShop/PrestaShop/pull/13584/commits/b87457615bb71c93d4368028050b6beb0abcbd07 wdyt?

Reply:
See answer https://github.com/PrestaShop/PrestaShop/pull/13584/files#r301597349 😉